### PR TITLE
Switch to Kubernetes Community-Owned Package Repositories

### LINKED ISSUE #4087: Switch to Kubernetes Community-Owned Package Repositories
A [recent announcement](https://kubernetes.io/blog/2023/08/15/pkgs-k8s-io-introduction/) specifies that we should be using pkgs.k8s.io for all k8s artifact downloads.

There are several references to `https://packages.cloud.google.com` and `https://apt.kubernetes.io` in templates in the codebase, specifically the ones that download unreleased k8s CI artifacts: https://github.com/search?q=repo%3Akubernetes-sigs%2Fcluster-api-provider-azure+packages.cloud.google.com&type=code

We should use pkgs.k8s.io for Kubernetes ecosystem artifacts, following this guidance: https://kubernetes.io/blog/2023/08/15/pkgs-k8s-io-introduction/

/kind feature

### PR DESCRIPTION
**Pull request by willie-yao**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
A [recent announcement](https://kubernetes.io/blog/2023/08/15/pkgs-k8s-io-introduction/) specifies that we should be using pkgs.k8s.io for all k8s artifact downloads. This PR replaces references to `https://packages.cloud.google.com` and `https://apt.kubernetes.io` in templates in the codebase with `pkgs.k8s.io`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4087 

**Special notes for your reviewer**:

Changes are made according to the migration guide: https://kubernetes.io/blog/2023/08/15/pkgs-k8s-io-introduction/#how-to-migrate

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Use pkgs.k8s.io
```

### CODE REVIEW

File: templates/test/ci/cluster-template-prow-ci-version.yaml [hunk 1]
```diff
@@ -125,8 +125,8 @@ spec:
           if [[ "$${CI_VERSION}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
             VERSION_WITHOUT_PREFIX="${CI_VERSION#v}"
             DEBIAN_FRONTEND=noninteractive apt-get install -y apt-transport-https curl
-            curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
-            echo 'deb https://apt.kubernetes.io/ kubernetes-xenial main' > /etc/apt/sources.list.d/kubernetes.list
+            curl -fsSL https://pkgs.k8s.io/core:/stable:/${KUBERNETES_VERSION}/deb/Release.key | sudo gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg
+            echo "deb [signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg] https://pkgs.k8s.io/core:/stable:/${KUBERNETES_VERSION}/deb/ /" | sudo tee /etc/apt/sources.list.d/kubernetes.list
```
Comment (mboersma) [inline templates/test/ci/cluster-template-prow-ci-version.yaml:129]: Reading a few lines down, I came across `apt-cache madison` which I'd never seen before. I guess it's just a specialized version of `showpkg` that lives on: https://unix.stackexchange.com/questions/276037/why-apt-madison. TIL!

File: templates/test/ci/prow-machine-pool-ci-version/patches/machine-pool-ci-version.yaml [hunk 2]
```diff
@@ -35,8 +35,8 @@ spec:
           if [[ "$${CI_VERSION}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
             VERSION_WITHOUT_PREFIX="${CI_VERSION#v}"
             DEBIAN_FRONTEND=noninteractive apt-get install -y apt-transport-https curl
-            curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
-            echo 'deb https://apt.kubernetes.io/ kubernetes-xenial main' > /etc/apt/sources.list.d/kubernetes.list
+            curl -fsSL https://pkgs.k8s.io/core:/stable:/${KUBERNETES_VERSION}/deb/Release.key | sudo gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg
```
Comment (CecileRobertMichon) [inline templates/test/ci/prow-machine-pool-ci-version/patches/machine-pool-ci-version.yaml:38]: these scripts are running in cloud-init so already running as root, we shouldn't need to use `sudo` anywhere

**Pull request merged**